sky/serve/controller.py

```diff
@@ -119,7 +122,21 @@ async def update_service(request: fastapi.Request):
                     f'Update to new version version {version}: {service}')
 
                 self._replica_manager.update_version(version, service)
-                self._autoscaler.update_version(version, service)
+
+                # TODO(MaoZiming): use NAME for autoscaler.
+                if not (isinstance(
+                        self._autoscaler,
+                        type(
+                            autoscalers.RequestRateAutoscaler.from_spec(
+                                service)))):
+                    old_autoscaler = self._autoscaler
+                    self._autoscaler = (
+                        autoscalers.RequestRateAutoscaler.from_spec(service))
+                    self._autoscaler.request_timestamps = (
+                        old_autoscaler.request_timestamps)
+                    self._autoscaler.latest_version = version
```

Comment:
Again, add a todo https://github.com/skypilot-org/skypilot/pull/2793/files#r1489108818

Reply:
What is this todo?

Reply:
added

Reply:
We just need to copy over the `request_timestamps`? 
We do not want to inherit all the states from the previous autoscaler, as the new autoscaler is instantiate from the service spec anyway.

Reply:
added. (offline discussed)